core/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -418,7 +419,7 @@ static Query parseQueryString(ExtendedCommonTermsQuery query, Object queryString
         }
 
         if (count == 0) {
-            return null;
+            return new MatchNoDocsQuery("Zero tokens in analyzed query.");
         }
```

Comment:
Looks like CommonTermsQuery rewrites empty terms clauses to MatchNoDocsQuery(). So I will remove this whole check.